paymentsheet/src/main/java/com/stripe/android/customersheet/data/CustomerSessionElementsSessionManager.kt

```diff
@@ -87,16 +97,59 @@ internal class DefaultCustomerSessionElementsSessionManager @Inject constructor(
                         clientSecret = customerSessionClientSecret.clientSecret,
                     ),
                     externalPaymentMethods = listOf(),
-                ).onSuccess { elementsSession ->
-                    elementsSession.customer?.session?.run {
-                        cachedCustomerEphemeralKey = CachedCustomerEphemeralKey.Available(
-                            customerId = customerId,
-                            ephemeralKey = apiKey,
-                            expiresAt = apiKeyExpiry,
+                ).mapCatching { elementsSession ->
+                    val customer = elementsSession.customer ?: run {
+                        errorReporter.report(
+                            ErrorReporter
+                                .UnexpectedErrorEvent
+                                .CUSTOMER_SESSION_ON_CUSTOMER_SHEET_ELEMENTS_SESSION_NO_CUSTOMER_FIELD
+                        )
+
+                        throw IllegalStateException(
+                            "`customer` field should be available when using `CustomerSession` in elements/session!"
                         )
                     }
+
+                    ElementsSessionWithCustomer(
+                        elementsSession = elementsSession,
+                        customer = customer
+                    )
+                }.onSuccess { elementsSessionWithCustomer ->
+                    val customerSession = elementsSessionWithCustomer.customer.session
+
+                    cachedCustomerEphemeralKey = CachedCustomerEphemeralKey.Available(
+                        customerId = customerSession.customerId,
+                        ephemeralKey = customerSession.apiKey,
+                        expiresAt = customerSession.apiKeyExpiry,
+                    )
                 }.getOrThrow()
             }
         }
     }
+
+    private fun validateCustomerSessionClientSecret(customerSessionClientSecret: String) {
```

Comment:
is there a way we can share this code with payment sheet rather than re-implementing it?

Reply:
Added a `CustomerSessionClientSecretProvider` that does the actual validation. Error messages are still different between both implementations given the context in how the errors are produced is different between both `PaymentSheet` and `CustomerSheet`.

Reply:
great, thanks!